.github/workflows/_test_rosetta.yaml

```diff
@@ -14,16 +14,52 @@ on:
 
 jobs:
   rosetta-tests:
-    runs-on: ubuntu-22.04
+    strategy:
+      matrix:
+        MARKERS: ["", "-m integration"]
+      fail-fast: false
+    runs-on: [self-hosted, compute, V100]
     steps:
       - name: Print environment variables
         run: |
           env
 
+      - name: Print GPU information
+        run: nvidia-smi  
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Pull Rosetta image
+        shell: bash -x -e {0}
+        run: |
+          docker pull ${{ inputs.ROSETTA_IMAGE }}
+
+      - name: Run Rosetta tests w/ docker
+        run: |
+          set -x
+          docker run --gpus '"device=0,1"' ${{ inputs.ROSETTA_IMAGE }} sh -c "pip install '/opt/rosetta[test]' && pytest /opt/rosetta ${{ matrix.MARKERS }}"
```

Comment:
Is there a particular reason for only using 2 GPUs? Preferably, we don't hardcode the number of GPUs since the compute nodes may have a variable number of GPUs available.

Reply:
So my assumption here was that the tests could always assume there were 2 GPUs so that we wouldn't have to create additional logic to select the right devices. The example in mind, was if someone pjitt-ed a function for DP, the batch size would need to be divisible by the number of devices. So if the tests had arrays with BS=2, we'd be okay if num_devices<=2, but if num_devices>2 we'd error.

My understanding was that after jax was imported, the "visible devices" was fixed, so to run a test on a subset of GPUs, we'd need to run in a subprocess. Is there a way to make devices after jax has been imported? If so, I agree with what you're suggesting.

Reply:
Wellll, right now some tests do have hard coded batch sizes, but I suppose they could be refactored so that:
1. odd number of devices > 1 throws an error and
2. batchsize should be a multiple of 8

and then we should be robust to most common configurations.

I can remove the explicit device specification for now since there aren't any "real" tests anyways, but does the above sound good?

Reply:
Sounds good.

Reply:
Done